src/test/java/io/jenkins/plugins/grading/CoverageScoreTests.java

```diff
@@ -14,15 +12,74 @@
-        CoverageScore coverageScore = new CoverageScore(coverageConfiguration, Ratio.create(99, 100));
+    void shouldCalculateTotalImpactWithZeroCoveredImpact() {
+        final CoverageConfiguration coverageConfiguration = createCoverageConfiguration(-2, 0);
+        final CoverageScore coverageScore = new CoverageScore(coverageConfiguration, Ratio.create(99, 100));
 
         assertThat(coverageScore).hasTotalImpact(-2);
     }
+
+    @Test
+    void shouldCalculateTotalImpactWithZeroMissedImpact() {
+        final CoverageConfiguration coverageConfiguration = createCoverageConfiguration(0, 5);
+        final CoverageScore coverageScore = new CoverageScore(coverageConfiguration, Ratio.create(99, 100));
+
+        assertThat(coverageScore).hasTotalImpact(495);
+    }
+
+    @Test
+    void shouldCalculateTotalImpact() {
+        final CoverageConfiguration coverageConfiguration = createCoverageConfiguration(-1, 3);
+        final CoverageScore coverageScore = new CoverageScore(coverageConfiguration, Ratio.create(99, 100));
+
+        assertThat(coverageScore).hasTotalImpact(296);
+    }
+
+    @Test
+    void shouldGetId() {
+        final CoverageConfiguration coverageConfiguration = new CoverageConfigurationBuilder().build();
+        final CoverageScore coverageScore = new CoverageScore(coverageConfiguration, Ratio.create(99, 100));
+
+        assertThat(coverageScore.getId()).isEqualTo("Line");
+    }
+
+    @Test
+    void shouldGetTotalImpact() {
+        final CoverageConfiguration coverageConfiguration = createCoverageConfiguration(1, 1);
+        final CoverageScore coverageScore = new CoverageScore(coverageConfiguration, Ratio.create(99, 100));
+
+        final int missedImpact = coverageConfiguration.getMissedImpact() * coverageScore.getMissedSize();
+        final int coveredImpact = coverageConfiguration.getCoveredImpact() * coverageScore.getCoveredSize();
+        assertThat(coverageScore.getTotalImpact()).isEqualTo(missedImpact + coveredImpact);
+    }
+
+    @Test
+    void shouldGetCoveredSize() {
+        final Ratio codeCoverageRatio = Ratio.create(99, 100);
+        final CoverageConfiguration coverageConfiguration = new CoverageConfigurationBuilder().build();
+        final CoverageScore coverageScore = new CoverageScore(coverageConfiguration, codeCoverageRatio);
+
+        assertThat(coverageScore.getCoveredSize()).isEqualTo(codeCoverageRatio.getPercentage());
+    }
+
+    @Test
+    void shouldGetMissedSize() {
+        final Ratio codeCoverageRatio = Ratio.create(99, 100);
+        final CoverageConfiguration coverageConfiguration = new CoverageConfigurationBuilder().build();
+        final CoverageScore coverageScore = new CoverageScore(coverageConfiguration, codeCoverageRatio);
+
+        assertThat(coverageScore.getMissedSize()).isEqualTo(100 - codeCoverageRatio.getPercentage());
+    }
+
+    private CoverageConfiguration createCoverageConfiguration(final int missedImpact, final int coveredImpact) {
```

Comment:
Geht damit nicht genau die Nützlichkeit des Builders verloren?

Reply:
Ja das ist richtig. Allerdings enstehen keine Vorteile, wenn ich den Builder inline in jeder Testmethode verwende, oder?
Der Test wird dadurch eigentlich nur länger und wie genau ein CoverageConfiguration Objekt erstellt wird, ist für den Leser des Tests nicht wichtig. Deshalb habe ich mich dafür entschieden die Erzeugung der CoverageConfiguration zu abstrahieren. So reduziere ich außerdem die Abhängigkeiten zum Konstruktor/Builder und muss nicht jede Testmethode einzeln anfassen, wenn sich etwas an der Erzeugung von CoverageConfiguration ändert, sondern nur diese eine Creation Method.